component/src/main/java/com/synopsys/integration/alert/component/authentication/security/saml/SAMLManager.java

```diff
@@ -166,4 +173,33 @@ private ExtendedMetadataDelegate createDelegate(MetadataProvider provider) {
     private HttpClient httpClient() {
         return new HttpClient(new MultiThreadedHttpConnectionManager());
     }
+
+    private boolean checkMetadataProvidersValid(String metaDataUrl, String entityId) {
+        boolean httpProviderValid = false;
+        boolean fileProviderValid = false;
+        boolean entityIdValid = StringUtils.isNotBlank(entityId);
+        if (!entityIdValid) {
+            logger.error("Validated SAML entity id missing.");
+        }
+        try {
+            Optional<ExtendedMetadataDelegate> provider = createHttpProvider(metaDataUrl);
+            if (provider.isPresent()) {
+                provider.get().initialize();
+                httpProviderValid = true;
+            }
+        } catch (Exception ex) {
```

Comment:
Should we be handling such broad exceptions here?

Reply:
Yes, because if there's a runtime exception on initialize of the metadata provider then you won't be able to login with SAML, LDAP or the database users.  The only exception that is advertised by the initialize method is a MetadataProviderException, but if the XML Metadata file isn't formatted correctly you can get a SAXParser exception which isn't wrapped in a MetaDataProviderException or advertised as a thrown exception.  We don't want the exception to bubble all the way up because it could cause the server to crash.

Here we are validating as best we can that the SAML configuration is good.  If it is, then configure SAML.  If it isn't valid, then we want to be able to disable the SAML configuration.  SAML has two parts to it.  The database configuration to enable SAML authentication in Alert, which has already happened before this point, and the SAML library portion.  We need to be able to recover from an exception in order to disable SAML configuration in the database. This is the most cautious way to be able to recover.